### PR TITLE
Drop gRPC server side custom codegen

### PR DESCRIPTION
**Pull request by nmarton-da**
[CHANGELOG_BEGIN]
[CHANGELOG_END]

### CODE REVIEW

**Comment by daravep** [comment]
@nmarton-da Stupid question: why has this akka stuff been added at all then? It seems that you are able to drop this quite complicated & special build thing by just changing 80 lines of code. So why did we add it in the first place? Do you have an idea?

**Comment by stefanobaghino-da** [comment]
Historical context (quite opaque):
- https://digitalasset.atlassian.net/browse/DEL-6794
- https://github.com/DACH-NY/da/pull/18614

**Comment by tudor-da** [comment]
Fixed a couple of small nits.

**Comment by tudor-da** [comment]
> Historical context (quite opaque):
> 
> * https://digitalasset.atlassian.net/browse/DEL-6794
> * [DEL-6794: ledger-bindings: akka streams codegen DACH-NY/da#18614](https://github.com/DACH-NY/da/pull/18614)

Thanks for the refs, @stefanobaghino-da . Do you know more than what meets the eye in there? It looks to me like this codegenc was more like syntactic sugar than necessity. Anyway, I think we can remove

**Comment by stefanobaghino-da** [comment]
> > Historical context (quite opaque):
> > 
> > * https://digitalasset.atlassian.net/browse/DEL-6794
> > * [DEL-6794: ledger-bindings: akka streams codegen DACH-NY/da#18614](https://github.com/DACH-NY/da/pull/18614)
> 
> Thanks for the refs, @stefanobaghino-da . Do you know more than what meets the eye in there? It looks to me like this codegen was more like syntactic sugar than necessity. Anyway, I think we can remove

The PR mentions closing all streams when `close()` is called. Not 100% sure of the meaning.

**Comment by tudor-da** [comment]
> The PR mentions closing all streams when `close()` is called. Not 100% sure of the meaning.

Yeah noticed that, but couldn't figure further. Anyway. The `StreamingServiceLifecycleManagement` already takes care of closing streams so we're fine

**Review by daravep — COMMENTED** [review]
I'll leave the detailed review to @mziolekda but I really appreciate this simplification!

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/GrpcTransactionService.scala [hunk 1]
```diff
@@ -56,9 +59,12 @@ final class GrpcTransactionService(
     }
   }
 
-  override protected def getTransactionTreesSource(
-      request: GetTransactionsRequest
-  ): Source[GetTransactionTreesResponse, NotUsed] = {
+  def getTransactionTrees(
+      request: com.daml.ledger.api.v1.transaction_service.GetTransactionsRequest,
```
Comment (mziolekda) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/GrpcTransactionService.scala:63]: what happened here?

**Review by mziolekda — APPROVED** [review]
Thanks @nmarton-da and @tudor-da. It looks good!

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/GrpcTransactionService.scala (hunk 1 above)
Comment (tudor-da) [reply ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/GrpcTransactionService.scala]: thanks, missed it

**Pull request merged**